### PR TITLE
docs: cosmetic changes to `Database` spec

### LINKED ISSUE #6278: [Bug]: Fix API reference for `Database` CRD
### Is there an existing issue already for this bug?

- [x] I have searched for an existing issue, and could not find anything. I believe this is a new bug.

### I have read the troubleshooting guide

- [x] I have read the troubleshooting guide and I think this is a new bug.

### I am running a supported version of CloudNativePG

- [x] I have read the troubleshooting guide and I think this is a new bug.

### Contact Details

_No response_

### Version

trunk (main)

### What version of Kubernetes are you using?

1.31

### What is your Kubernetes environment?

Self-managed: kind (evaluation)

### How did you install the operator?

YAML manifest

### What happened?

The API reference doesn't look good for the `Database` CRD.

### Cluster resource

```shell

```

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by gbartolini**
Closes #6278

### CODE REVIEW

**Review by mnencia — APPROVED** [review]

File: api/v1/database_types.go [hunk 1]
```diff
@@ -40,105 +40,114 @@ const (
...
 	BuiltinLocale string `json:"builtinLocale,omitempty"`
 
-	// The COLLATION_VERSION (cannot be changed)
+	// Maps to the `COLLATION_VERSION` parameter of `CREATE DATABASE`. This setting cannot be changed.
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="collationVersion is immutable"
 	// +optional
 	CollationVersion string `json:"collationVersion,omitempty"`
 
-	// True when the database is a template
+	// Maps to the `IS_TEMPLATE` parameter of `CREATE DATABASE`.
```
Comment (jsilvela) [inline api/v1/database_types.go:126]: On this one and other re-settable parameters such as CONNECTION LIMIT.
I don't love that we say "Maps to ... parameter of `CREATE DATABASE`"
They also apply to ALTER.
What about just saying "Maps to  ... parameter for databases inPostgres"

File: api/v1/database_types.go [hunk 2]
```diff
@@ -40,105 +40,114 @@ const (
...
-	// The name inside PostgreSQL
+	// The name of the database to create inside PostgreSQL. This setting cannot be changed.
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
 	// +kubebuilder:validation:XValidation:rule="self != 'postgres'",message="the name postgres is reserved"
 	// +kubebuilder:validation:XValidation:rule="self != 'template0'",message="the name template0 is reserved"
 	// +kubebuilder:validation:XValidation:rule="self != 'template1'",message="the name template1 is reserved"
 	Name string `json:"name"`
 
-	// The owner
+	// The role name of the user who own the database inside PostgreSQL.
```
Comment (jsilvela) [inline api/v1/database_types.go:59]:
```suggestion
	// The role name of the user who owns the database inside PostgreSQL.
```

File: api/v1/database_types.go (hunk 1 above)
Comment (gbartolini) [reply api/v1/database_types.go:126]: Good idea. Let me try.

**Review by jsilvela — APPROVED** [review]

**Pull request merged**